qiskit_nature/second_q/properties/electronic_density.py

```diff
@@ -33,6 +34,109 @@ class ElectronicDensity(ElectronicIntegrals):
     :meth:`~qiskit_nature.second_q.operators.ElectronicIntegrals.trace_spin`.
     """
 
+    @staticmethod
+    def _rdm2_from_rdm1s(
+        rdm1_a: np.ndarray, rdm1_b: np.ndarray, *, mixed_spin: bool = False
+    ) -> np.ndarray:
+        einsum_func, _ = get_einsum()
+        rdm2 = einsum_func("ij,kl->ijkl", rdm1_a, rdm1_b)
+        if not mixed_spin:
+            rdm2 -= einsum_func("ij,kl->ikjl", rdm1_a, rdm1_b)
+
+        return rdm2
+
+    @classmethod
+    def empty(cls, num_spatial_orbitals: int, *, include_rdm2: bool = True) -> ElectronicDensity:
+        """Initializes an empty (all-zero) ``ElectronicDensity``.
+
+        Args:
+            num_spatial_orbitals: the number of spatial orbitals.
+            include_rdm2: whether to include the 2-body RDMs.
+
+        Returns:
+            The resulting ``ElectronicDensity``.
+        """
+        rdm1 = np.zeros((num_spatial_orbitals, num_spatial_orbitals), dtype=float)
+        rdm2 = (
+            np.zeros(
+                (
+                    num_spatial_orbitals,
+                    num_spatial_orbitals,
+                    num_spatial_orbitals,
+                    num_spatial_orbitals,
+                ),
+                dtype=float,
+            )
+            if include_rdm2
+            else None
+        )
+
+        return ElectronicDensity.from_raw_integrals(
+            rdm1, rdm2, rdm1, rdm2, rdm2, auto_index_order=False
+        )
+
+    @classmethod
+    def identity(cls, num_spatial_orbitals: int, *, include_rdm2: bool = True) -> ElectronicDensity:
+        """Initializes an identity ``ElectronicDensity``.
+
+        Args:
+            num_spatial_orbitals: the number of spatial orbitals.
+            include_rdm2: whether to include the 2-body RDMs.
+
+        Returns:
+            The resulting ``ElectronicDensity``.
+        """
+        rdm1 = np.eye(num_spatial_orbitals, dtype=float)
+        rdm2: np.ndarray | None = None
+        rdm2_ba: np.ndarray | None = None
+        if include_rdm2:
+            rdm2 = ElectronicDensity._rdm2_from_rdm1s(rdm1, rdm1)
+            rdm2_ba = ElectronicDensity._rdm2_from_rdm1s(rdm1, rdm1, mixed_spin=True)
+
+        return ElectronicDensity.from_raw_integrals(
+            rdm1, rdm2, rdm1, rdm2, rdm2_ba, auto_index_order=False
+        )
+
+    @staticmethod
+    def from_particle_number(
```

Comment:
Good question :sweat_smile: I do not recall a good reason why I would have made `from_orbital_occupation` a `staticmethod`. Probably because it does not use `cls` (although it should use that instead of hard-coding the `ElectronicDensity` to allow subclassing)..
The reason `empty` and `identity` are `classmethod` is because I mirrored this from `SparsePauliOp.zero` and `.one`, resp.

Maybe we can change this to be `classmethod` and at the same time change `from_orbital_occupation` to be a `classmethod`, too? I'm not sure about deprecation-implications of the latter though.. :thinking:

Reply:
In fact one can simply call a `classmethod` also from an instance. That means nothing effectively changes from the user side of thins so I updated the code to use `classmethod` in both cases :+1: